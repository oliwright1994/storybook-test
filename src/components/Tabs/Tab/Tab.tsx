import { Tab as ChakraTab, TabPanelProps as IChakraTab } from '@chakra-ui/core'
import React, { FC } from 'react'

const Tab: FC<IChakraTab> = React.forwardRef(({ children, ...restProps }, ref) => {
  return (
    <ChakraTab ref={ref} {...restProps}>
      {children}
    </ChakraTab>
  )
})

export default Tab
