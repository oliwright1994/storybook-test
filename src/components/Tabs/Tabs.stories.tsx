import createStory from '@bright/storybook/createStory'
import React from 'react'

import Tabs from '.'

createStory('COMPONENTS | Tabs', [
  {
    label: 'Tabs',
    jsx: () => {
      return (
        <Tabs>
          <Tabs.TabList>
            <Tabs.Tab>Tab1</Tabs.Tab>
            <Tabs.Tab>Tab2</Tabs.Tab>
            <Tabs.Tab>Tab2</Tabs.Tab>
          </Tabs.TabList>
          <Tabs.TabPanels>
            <Tabs.TabPanel>Panel One</Tabs.TabPanel>
            <Tabs.TabPanel>Panel Two</Tabs.TabPanel>
            <Tabs.TabPanel>Panel Three</Tabs.TabPanel>
          </Tabs.TabPanels>
        </Tabs>
      )
    },
  },
])
