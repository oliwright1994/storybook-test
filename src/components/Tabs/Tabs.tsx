import { ITabs as IChakraTabs, Tabs as ChakraTabs } from '@chakra-ui/core'
import React, { FC } from 'react'

import Tab from './Tab'
import TabList from './TabList'
import TabPanel from './TabPanel'
import TabPanels from './TabPanels'

interface ITabs extends FC<IChakraTabs> {
  Tab: typeof Tab
  TabList: typeof TabList
  TabPanel: typeof TabPanel
  TabPanels: typeof TabPanels
}

const Tabs: ITabs = ({ children }) => {
  return <ChakraTabs  flexGrow={1}>{children}</ChakraTabs>
}

Tabs.Tab = Tab
Tabs.TabList = TabList
Tabs.TabPanel = TabPanel
Tabs.TabPanels = TabPanels

export default Tabs
