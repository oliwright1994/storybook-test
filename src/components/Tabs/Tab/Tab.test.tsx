import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Tabs from '../Tabs'
import Tab from '.'

describe('<Tab />', () => {
  const props = {
    children: 'Tab1',
  }
  const tabsComponent = (
    <Tabs>
      <Tabs.TabList>
        <Tab {...props} />
      </Tabs.TabList>
    </Tabs>
  )

  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(tabsComponent)
    expect(container).toMatchSnapshot()
  })

  it('should display text', () => {
    const { container } = renderWithBrightTheme(tabsComponent)
    expect(container).toHaveTextContent(props.children)
  })
})
