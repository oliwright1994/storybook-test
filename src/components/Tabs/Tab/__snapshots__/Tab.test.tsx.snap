// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> should render as expected 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      aria-orientation="horizontal"
      class="css-1dxxh68"
      role="tablist"
    >
      <button
        aria-controls="panel:1-0"
        aria-selected="true"
        class="css-1kt89oh"
        id="tab:1-0"
        role="tab"
        tabindex="0"
        type="button"
      >
        Tab1
      </button>
    </div>
  </div>
</div>
`;
